hbs-helpers.js

```diff
@@ -22,8 +22,8 @@ function localizedBreachDataClasses(supportedLocales, dataClasses, args) {
   const localizedDataClasses = [];
   if (dataClasses.constructor === Array) {
     dataClasses.forEach(dataClass => {
-      dataClass = dataClass.replace("'", "");
-      localizedDataClasses.push(LocaleUtils.fluentFormat(supportedLocales, dataClass.split(" ").join("-").toLowerCase(), args));
+      dataClass = dataClass.replace(/[^-a-z0-9]/ig,"-").toLowerCase();
```

Comment:
I feel like this will be needed on L30 below as well, and maybe a candidate for a new function maybe. Not sure if we care about how pretty the string is with multiple concurrent hyphens or trailing hyphens. Maybe not worth extra fanciness if it’s only an l10n key.

Reply:
Here's a pretty crude `slugify()`, if it helps:

```js
function slugify(str) {
  return str.toLowerCase()
    .replace(/[^-a-z0-9]/ig, "-") // replace non a-z and 0-9 w/ hyphens. 
    .replace(/-{2,}/g, "-") // collapse multiple hyphens into single hyphen.
    .replace(/^\-?(.+?)\-?$/g, "$1"); // trim any leading or trailing hyphens.
}

console.log(slugify("Peter is lame")); // peter-is-lame
console.log(slugify("Peter is as lame (as a duck)")); // peter-is-as-lame-as-a-duck
console.log(slugify("Peter's lameness intensifies!")); // peter-s-lameness-intensifies
console.log(slugify("'(fancy) pork', they shouted!")); // fancy-pork-they-shouted
console.log(slugify("----peter-IS-LAAAMMMMEEEE---- and that's a guarantee!!!")); // peter-is-laaammmmeeee-and-that-s-a-guarantee
```

Reply:
Actually, here's the diff from the current parser [1] versus the "new" `slugify` parser [2]. But basically this was just an excuse for me to test out **fluent-syntax** for parsing .ftl files.

```js
const {readFileSync} = require("fs");
const {parse} = require("fluent-syntax");

const dataClasses = readFileSync("./locales/en/data-classes.ftl", "utf-8");

parse(dataClasses).body.forEach(item => {
  if (item.type === "GroupComment") return;
  const value = item.value.elements.map(({value}) => value).join(" ");
  const slug1 = item.id.name;
  const slug2 = slugify(value);
  if (slug1 !== slug2) {
    console.log(`${value}\n\t[1]: ${slug1}\n\t[2]: ${slug2}`);
  }
});

function slugify(str) {
  return str.toLowerCase()
    .replace(/[^-a-z0-9]/g, "-")
    .replace(/-{2,}/g, "-")
    .replace(/(^-|-$)/g, "");
}
```

```
Family members’ names
        [1]: family-members--names
        [2]: family-members-names

School grades (class levels)
        [1]: school-grades--class-levels-
        [2]: school-grades-class-levels
```

Reply:
Thanks @pdehaan ! Got mired in email stuff yesterday and missed these until much later. They're just l10n keys, but I think it's worth getting rid of trailing / multiple consecutive hyphens anyway. Just pushed up something to format the l10n keys when we load breaches, but it doesn't address what happens when new data classes are added.